manifests/passwd_check.pp

```diff
@@ -0,0 +1,32 @@
+# @summary Verify that passwd services (e.g., sssd) are working.
+# The primary use case is to ensure that sssd is configured
+# and running prior to starting slurmctld so we do not lose
+# reservations or prevent users from submitting jobs (both of
+# which require resolution of usernames and potentially groups).
+#
+# Verify that passwd services (e.g., sssd) are working.
+#
+# @param check_users
+#   Users to check for to verify functional passwd services.
+#   CURRENTLY ONLY SUPPORTS AN ARRAY OF ONE USER!
+#
+# @example
+#   include profile_slurm::passwd_check
+class profile_slurm::passwd_check (
+
+  Array[String, 1, 1] $check_users,
```

Comment:
This is confusing if it only supports an array with 1 element.

Any reason to not just make this be a string until an array of users is possible?

Reply:
I was just thinking we might want to check multiple users down the line, but not implementing that capability yet, and so trying to avoid breakage later (although we could support both String and Array and have Puppet detect what the user inputs).

On the other hand, after sitting on it for a bit, I can't think of a good reason to check multiple users or multiple groups. E.g., if checking groups:
* Would we code the check to succeed as long as it finds at least one of the groups? It seems like if even one group is not found that's a potential problem and the check should fail.
* Or would we code the check to succeed only if ALL of the groups are found? I'm not sure that spot checking additional groups provides much more assurance, and it might just make the check take longer and the code be more complicated. I don't think it's worth it.

So I'll just make it a String (and change it to a group instead of a user).